server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -261,10 +260,10 @@ private Metadata(
         this.persistentSettings = persistentSettings;
         this.settings = settings;
         this.hashesOfConsistentSettings = hashesOfConsistentSettings;
-        this.indices = indices;
-        this.aliasedIndices = aliasedIndices;
-        this.customs = customs;
-        this.templates = templates;
+        this.indices = Map.copyOf(indices);
```

Comment:
These are copies to make sure the maps in this object are all immutable, but this constructor is private - so we could instead just make sure all calls to the constructor are done using immutablemaps (`Map.copyOf` checks the type and just casts if the arg is already immutable...). What's the approach I should take here?